dotCMS/src/integration-test/java/com/dotcms/publishing/PublisherAPITest.java

```diff
@@ -319,4 +326,291 @@ private FileFilter getNoBundleXMLFileFilter() {
+    }
+
+    /**
+     * This test creates 2 FilterDescriptors (one set as default) tries to get a FilterDescriptor, but since key is not passed,
+     * it returns the FilterDescriptor set as default.
+     */
+    @Test
+    public void test_getFilterDescriptorByKey_emptyKey_returnDefaultFilter(){
+        publisherAPI.getFilterDescriptorMap().clear();
+
+        final String keyDefault = "TestByKeyEmptyKeyDefault.yml";
+        createFilterDescriptor(keyDefault,"TestByKeyEmptyKeyDefault",true,null);
+
+        final String keyNonDefault = "TestByKeyEmptyKeyNonDefault.yml";
+        createFilterDescriptor(keyNonDefault,"TestByKeyEmptyKeyDefault",false,null);
+
+        final FilterDescriptor filterDescriptor = publisherAPI.getFilterDescriptorByKey("");
+        Assert.assertNotNull(filterDescriptor);
+        Assert.assertEquals(keyDefault,filterDescriptor.getKey());
+    }
+
+    /**
+     * This test creates 2 FilterDescriptors (one set as default) tries to get a FilterDescriptor, but since key passed does not belong to any FilterDescriptor,
+     * it returns the FilterDescriptor set as default.
+     */
+    @Test
+    public void test_getFilterDescriptorByKey_filterKeyDoesNotExist_returnDefaultFilter(){
+        publisherAPI.getFilterDescriptorMap().clear();
+
+        final String keyDefault = "TestByKeyDoesNotExistDefault.yml";
+        createFilterDescriptor(keyDefault,"TestByKeyDoesNotExistDefault",true,null);
+
+        final String keyNonDefault = "TestByKeyDoesNotExistNonDefault.yml";
+        createFilterDescriptor(keyNonDefault,"TestByKeyDoesNotExistNonDefault",false,null);
+
+        final FilterDescriptor filterDescriptor = publisherAPI.getFilterDescriptorByKey("thisKeyNotExists");
+        Assert.assertNotNull(filterDescriptor);
+        Assert.assertEquals(keyDefault,filterDescriptor.getKey());
+    }
+
+    /**
+     * This test creates a FilterDescriptor and a Bundle (with the created FilterDescriptor) and
+     * returns a PublisherFilter with the filters of the FilterDescriptor.
+     *
+     * In this case the FilterDescriptor has all the possible filters.
+     */
+    @Test
+    public void test_createPublisherFilter_withAllFilters()
+            throws DotDataException, DotSecurityException {
+        publisherAPI.getFilterDescriptorMap().clear();
+
+        final String filterKey = "TestCreatePublisherFilterAllFilters.yml";
+        final Map<String,Object> filtersMap = new HashMap<>();
+        filtersMap.put("excludeQuery","+baseType:5");
+        final List<Object> listExcludeClasses = new ArrayList();
+        listExcludeClasses.add("User");
+        listExcludeClasses.add("Host");
+        filtersMap.put("excludeClasses", listExcludeClasses);
+        filtersMap.put("dependencies",true);
+        filtersMap.put("excludeDependencyQuery","+baseType:7");
+        filtersMap.put("excludeDependencyClasses", listExcludeClasses);
+        filtersMap.put("forcePush",false);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String literal "forcePush" appears 6 times in this file; the first occurrence is on line 551](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5458951)